src/machine/machine_stm32.go

```diff
@@ -2,6 +2,102 @@
 
 package machine
 
+import "unsafe"
+
 // Peripheral abstraction layer for the stm32.
 
 type PinMode uint8
+
+// AltFunc represents the alternate function peripherals that can be mapped to
+// the GPIO ports. Since these differ by what is supported on the stm32 family
+// they are defined in the more specific files
+type AltFunc uint8
+
+// Peripheral operations sequence:
+//  1. Enable the clock to the alternate function.
+//  2. Enable clock to corresponding GPIO
+//  3. Attach the alternate function.
+//  4. Configure the input-output port and pins (of the corresponding GPIOx) to match the AF .
+//  5. If desired enable the nested vector interrupt control to generate interrupts.
+//  6. Program the AF/peripheral for the required configuration (eg baud rate for a USART) .
+
+// Given that the stm32 family has the AF and GPIO on different registers based on the chip,
+//  use the main function here for configuring, and use hooks in the more specific chip
+//  definition files
+// Also, the stm32f1xx series handles things differently from the stm32f0/2/3/4
+
+// EnableClock configures the STM32 to send clock signals to the given GPIO port
+func (p Pin) EnableClock() {
```

Comment:
Why is this exported? It's already done in `Configure`, so there's no need to repeat it anywhere.

Reply:
Along the same lines as the `EnableAltFunc()`, providing it as a building block for developers to add stuff external to the codebase without needing to write the lower level code directly when they wanted to use the GPIO.